src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java

```diff
@@ -0,0 +1,126 @@
+package mate.academy.internetshop.dao.jdbc;
+
+import mate.academy.internetshop.dao.BucketDao;
+import mate.academy.internetshop.dao.ItemDao;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.lib.Inject;
+import mate.academy.internetshop.model.Bucket;
+import mate.academy.internetshop.model.Item;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.log4j.Logger;
+
+@Dao
+public class BucketDaoJdbcImpl extends AbstractDao<Bucket> implements BucketDao {
+    private static Logger logger = Logger.getLogger(BucketDaoJdbcImpl.class);
+    @Inject
+    private static ItemDao itemDao;
+
+    public BucketDaoJdbcImpl(Connection connection) {
+        super(connection);
+    }
+
+    @Override
+    public Bucket add(Bucket bucket) {
+        String query = "INSERT INTO buckets (user_id) VALUES (?);";
+        try (PreparedStatement statement = connection.prepareStatement(query,
+                PreparedStatement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, bucket.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            long dbBucketId = 0L;
+            while (resultSet.next()) {
+                dbBucketId = resultSet.getLong("bucket_id");
+            }
+            for (Item item : bucket.getItems()) {
+                addItemToBucket(dbBucketId, item.getId());
+            }
+            bucket = new Bucket(dbBucketId, bucket.getItems(), bucket.getUserId());
+        } catch (SQLException e) {
+            logger.error("Can't add new bucket with id=" + bucket.getId(), e);
+        }
+        return bucket;
+    }
+
+    @Override
+    public Optional<Bucket> get(Long bucketId) {
+        String query = "SELECT user_id FROM buckets WHERE bucket_id=?;";
+        long userId = 0L;
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, bucketId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                userId = resultSet.getLong("user_id");
+            }
+        } catch (SQLException e) {
+            logger.error("Can't get user id by bucket id=" + bucketId, e);
+        }
+        List<Item> items = getItemsFromBucket(bucketId);
+        return Optional.of(new Bucket(bucketId, items, userId));
```

Comment:
better move this two lines in try block:
List<Item> items = getItemsFromBucket(bucketId);
return Optional.of( new Bucket(bucketId, items, userId))

and after catch return Optional.empty()